tsp-output/@azure-tools/typespec-csharp/azoai/AzureOpenAI/AzureChatClient.cs

```diff
@@ -9,29 +10,41 @@ internal class AzureChatClient : Chat
 {
     private readonly string _apiVersion;
 
-    // TODO/Note: doesn't need to be virtual -- derived client could also set it in the 
-    // constructor.  Is one approach better?  I guess setting it in the constructor
-    // requires making it settable which maybe we don't want.
-    protected override ModelReaderWriterOptions ModelReaderWriterOptions => new("AzureWire");
-
     internal AzureChatClient(ClientPipeline pipeline, ApiKeyCredential credential, Uri endpoint, string apiVersion)
         : base(pipeline, credential, endpoint)
     {
         _apiVersion = apiVersion;
     }
 
-    public override Task<ClientResult<CreateChatCompletionResponse>> CreateChatCompletionAsync(CreateChatCompletionRequest createChatCompletionRequest, CancellationToken cancellationToken = default)
+    public override async Task<ClientResult<CreateChatCompletionResponse>> CreateChatCompletionAsync(CreateChatCompletionRequest createChatCompletionRequest, CancellationToken cancellationToken = default)
     {
-        return base.CreateChatCompletionAsync(createChatCompletionRequest, cancellationToken);
+        Argument.AssertNotNull(createChatCompletionRequest, nameof(createChatCompletionRequest));
+
+        // Initialize Azure input model
+        AzureCreateChatCompletionRequest azureChatCompletionOptions = new(createChatCompletionRequest);
+
+        using BinaryContent content = azureChatCompletionOptions.ToBinaryBody();
+        ClientResult result = await CreateChatCompletionAsync(createChatCompletionRequest.Model.ToString(), content, context: default).ConfigureAwait(false);
+        return ClientResult.FromValue(AzureCreateChatCompletionResponse.FromResponse(result.GetRawResponse()), result.GetRawResponse());
     }
 
     public override ClientResult<CreateChatCompletionResponse> CreateChatCompletion(CreateChatCompletionRequest createChatCompletionRequest, CancellationToken cancellationToken = default)
     {
-        return base.CreateChatCompletion(createChatCompletionRequest, cancellationToken);
+        Argument.AssertNotNull(createChatCompletionRequest, nameof(createChatCompletionRequest));
+
+        // Initialize Azure input model
+        AzureCreateChatCompletionRequest azureChatCompletionOptions = new(createChatCompletionRequest);
+
+        using BinaryContent content = azureChatCompletionOptions.ToBinaryBody();
+        ClientResult result = CreateChatCompletion(createChatCompletionRequest.Model.ToString(), content, context: default);
+        return ClientResult.FromValue(AzureCreateChatCompletionResponse.FromResponse(result.GetRawResponse()), result.GetRawResponse());
     }
 
     public override Task<ClientResult> CreateChatCompletionAsync(BinaryContent content, RequestOptions context = null)
     {
+        // Note, that we can later remap the values from the 3rd party client format to the 
```

Comment:
it would be good to hide this one, but I don't think we can do it given we don't expose the subtype

Reply:
Yes, that makes sense.  If we did expose the subtype, though, I think is a clear violation of the Liskov substitution principle -- I think you told me a few weeks ago that if we hide base type methods on the subtype, it makes the abstraction confusing ... why could I call this method on the base client, and now I don't see that method in the intellisense when I'm looking at a subtype?  I dunno.